test/sharness/t0115-gateway-dir-listing.sh

```diff
@@ -13,14 +13,34 @@ test_description="Test directory listing (dir-index-html) on the HTTP gateway"
 
 test_expect_success "ipfs init" '
   export IPFS_PATH="$(pwd)/.ipfs" &&
-  ipfs init --profile=test > /dev/null
+  ipfs init --empty-repo --profile=test > /dev/null
 '
 
 test_launch_ipfs_daemon_without_network
 
+# HAMT-sharded directory test. We must execute this test first as we want to
+# start from an empty repository and count the exact number of refs we want.
+HAMT_CID=bafybeiggvykl7skb2ndlmacg2k5modvudocffxjesexlod2pfvg5yhwrqm
+HAMT_REFS_CID=bafybeicv4utmj46mgbpamvw2k6zg4qjs5yvspfnlxz2y6iuey5smjfcn4a
+HAMT_REFS_COUNT=963
+
+test_expect_success "hamt: import fixture with necessary refs" '
+  ipfs dag import ../t0115-gateway-dir-listing/hamt-refs.car &&
+  ipfs refs local | wc -l | tr -d " " > refs_count_actual &&
+  echo $HAMT_REFS_COUNT > refs_count_expected &&
+  test_cmp refs_count_expected refs_count_actual
+'
+
+test_expect_success "hamt: fetch directory from gateway in offline mode" '
+  curl --max-time 1 -sD - http://127.0.0.1:$GWAY_PORT/ipfs/$HAMT_CID/ > hamt_list_response &&
```

Comment:
I added 1 second limit for the directory listing timeout. Not exactly sure how this is gonna behave in the CI. We may want to increase this to 2 seconds if it becomes non-reliable.

Reply:
Ok, it seems that it passed on both Circle CI and GitHub in around 0.6 seconds - it should be fine then.

Reply:
See https://github.com/ipfs/kubo/commit/d23702bbf8513094ca445bea52742d952e80a8dc#r104476111